### DISCUSSION TITLE
Simple 3D : Geometry not getting created with some models.

### DISCUSSION THREAD

**Discussion by RameshwarShinde** [iTwin/itwinjs-core#3703] [Q&A] [answered]
Hi,

**The scenario is :** 

I have implemented simple 3d primitives creation code referring => https://www.itwinjs.org/sandboxes/iTwinPlatform/Simple%203d
And trying to create the geometry with another iTwin model in my project (where I've implemented cross-probing too).

Working fine for some models, but geometry not getting created for models like iTwin's tutorial model in cross probing example https://www.itwinjs.org/sandboxes/iTwinPlatform/Cross-Probing

Tried this with some other models also, but no luck.
Code side it's getting created, but not in viewer.

**Question is :**
1. What's wrong with my scenario?
2. Why 3D primitives' creation only working with some models & some are not?
3. Is there any criteria that 3d geometry should be created with particular type of model only?

Plz help. Thanks in advance.

**Comment by pmconne** [comment, marked as answer]
Your problem is that you are creating decoration graphics similar to those in the sandbox you linked, but in some iModels you don't see it being displayed in the viewport?
It sounds like you may be positioning the decorations at a location far away from the location of your iModel.
As always, if that doesn't help, provide more details. We're not going to be able to effectively debug your code without seeing your code.

**Reply by RameshwarShinde** [reply]
Maybe its getting positioned far. In that case, to verify the location of iModel, 
I have consoled iModel. Seems like location parameters are set to zero there (see the image).
![image](https://user-images.githubusercontent.com/74453268/171175760-9333955a-598c-4d8f-8f78-d6f076fec828.png)

Is that right where I'm finding the location data of iModel? If not, Plz tell the way how can I check the locations of both iModel and geometry created.

**Reply by pmconne** [reply]
[globalOrigin](https://www.itwinjs.org/reference/core-common/imodels/imodel/globalorigin/) is an optional offset applied to spatial coordinates stored in the iModel - it's not relevant here.

Your iModel's geometry is contained within the iModel's [project extents](https://www.itwinjs.org/reference/core-common/imodels/imodel/projectextents/). You'll want to draw your decoration graphics somewhere within those extents too. In the absence of a more meaningful location, try deawing them at the center of the project extents.

**Reply by RameshwarShinde** [reply]
Thanks a lot @pmconne. As tried with Transform lib to bring it to the center of project extents. And it's working fine now.
